database/migrations/20190821144135_users.js

```diff
@@ -0,0 +1,19 @@
+
+exports.up = function(knex) {
+    return knex.schema.createTable('users', users => {
+        users.increments(); 
+        users.string('username', 255).notNullable().unique();
+        users.string('password', 255).notNullable();
+        users.string('email', 255).notNullable().unique();
+        users.integer('gender').notNullable()
```

Comment:
Can you explain why gender data type is integer, and have you considered that a user might choose not to disclose their gender?